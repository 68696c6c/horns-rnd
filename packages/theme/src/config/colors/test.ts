import Color from 'color'

import { Mode, makeColors } from '.'

describe('Colors', () => {
  it('should use dark background colors if mode is set to "dark"', () => {
    const config = makeColors({ mode: Mode.Dark })
    const result = config.background.base.base
    expect(Color(result).isDark()).toBe(true)
  })

  it('should use light background colors if mode is set to "light"', () => {
    const config = makeColors({ mode: Mode.Light })
    const result = config.background.base.base
    expect(Color(result).isDark()).toBe(false)
  })
})
