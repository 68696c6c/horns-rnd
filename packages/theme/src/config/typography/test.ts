import { makeTypography, Typography } from '.'

describe('makeTypography', () => {
  let result: Typography
  beforeEach(() => {
    result = makeTypography()
  })
  it('should match snapshot', () => {
    expect(result).toMatchSnapshot()
  })
  // it('should make the correct link styles', () => {
  //   console.log(result.link)
  //   // eslint-disable-next-line @typescript-eslint/no-use-before-define
  //   expect(result.link).toEqual(expectedLink)
  // })
})

// const expectedLink = {
//   active: {
//     align: 'left',
//     decoration: { line: 'underline', style: 'double' },
//     family: 'Helvetica, sans-serif',
//     indent: '',
//     kerning: 'normal',
//     letting: '',
//     size: 'medium',
//     spacing: '',
//     style: 'italic',
//     tracking: '',
//     transform: 'none',
//     weight: 400,
//   },
//   base: {
//     align: 'left',
//     decoration: { line: 'underline', style: 'solid' },
//     family: 'Helvetica, sans-serif',
//     indent: '',
//     kerning: 'normal',
//     letting: '',
//     size: 'medium',
//     spacing: '',
//     style: 'normal',
//     tracking: '',
//     transform: 'none',
//     weight: 400,
//   },
//   hover: {
//     align: 'left',
//     decoration: { line: 'underline', style: 'solid' },
//     family: 'Helvetica, sans-serif',
//     indent: '',
//     kerning: 'normal',
//     letting: '',
//     size: 'medium',
//     spacing: '',
//     style: 'italic',
//     tracking: '',
//     transform: 'none',
//     weight: 400,
//   },
//   inactive: {
//     align: 'left',
//     decoration: { line: 'underline', style: 'solid' },
//     family: 'Helvetica, sans-serif',
//     indent: '',
//     kerning: 'normal',
//     letting: '',
//     size: 'medium',
//     spacing: '',
//     style: 'normal',
//     tracking: '',
//     transform: 'none',
//     weight: 400,
//   },
//   visited: {
//     align: 'left',
//     decoration: { line: 'underline', style: 'solid' },
//     family: 'Helvetica, sans-serif',
//     indent: '',
//     kerning: 'normal',
//     letting: '',
//     size: 'medium',
//     spacing: '',
//     style: 'italic',
//     tracking: '',
//     transform: 'none',
//     weight: 400,
//   },
// }
