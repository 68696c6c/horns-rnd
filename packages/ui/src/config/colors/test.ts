import Color from 'color'

import { makeColors } from '.'

describe('Colors', () => {
  it('should use dark background colors if mode is set to "dark"', () => {
    const config = makeColors({ mode: 'dark' })
    const result = config.backgrounds.primary.base.base
    expect(Color(result).isDark()).toBe(true)
  })

  it('should use light background colors if mode is set to "light"', () => {
    const config = makeColors({ mode: 'light' })
    const result = config.backgrounds.primary.base.base
    expect(Color(result).isDark()).toBe(false)
  })
})
