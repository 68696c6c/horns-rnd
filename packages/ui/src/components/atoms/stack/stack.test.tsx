import React from 'react'
import { render } from '@testing-library/react'

import { theme } from '../../test'
import { Stack } from '.'

describe('Stack', () => {
  it('should render as default', () => {
    const { container } = render(
      <Stack theme={theme}>
        <div>one</div>
        <div>two</div>
        <div>three</div>
      </Stack>,
    )
    expect(container).toMatchSnapshot()
  })

  // TODO: figure out responsive testing
  // it('should render the items inline below the breakpoint', () => {
  //   const { container } = render(
  //     <Stack theme={theme} breakpoint={Breakpoint.Min}>
  //       <div>one</div>
  //       <div>two</div>
  //       <div>three</div>
  //     </Stack>,
  //   )
  //   expect(container).toMatchSnapshot()
  // })
})
