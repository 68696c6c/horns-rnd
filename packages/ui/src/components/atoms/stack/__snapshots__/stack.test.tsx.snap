// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack should render as default 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width: 480px) {
  .emotion-0 {
    display: grid;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div>
      one
    </div>
    <div>
      two
    </div>
    <div>
      three
    </div>
  </div>
</div>
`;
